ruma-api-macros/src/api/request.rs

```diff
@@ -49,9 +49,27 @@ impl Request {
 
             let field_name = &field.ident;
 
-            quote! {
-                #import_path::exports::http::header::#header_name,
-                #import_path::exports::http::header::HeaderValue::from_str(self.#field_name.as_ref())?
+            match &field.ty {
+                syn::Type::Path(syn::TypePath { path: syn::Path { segments, .. }, .. })
+                    if segments.iter().any(|seg| seg.ident == "Option") =>
```

Comment:
Uhh, I think we should use `path.is_ident("Option")` or `segments.last().unwrap() == "Option"`, not checking whether *any* segment is `Option`.